engine/src/main/java/org/camunda/bpm/engine/impl/metrics/parser/MetricsBpmnParseListener.java

```diff
@@ -18,25 +18,41 @@
 
 import org.camunda.bpm.engine.delegate.ExecutionListener;
 import org.camunda.bpm.engine.impl.bpmn.parser.AbstractBpmnParseListener;
+import org.camunda.bpm.engine.impl.persistence.entity.ExecutionEntity;
+import org.camunda.bpm.engine.impl.persistence.entity.ProcessDefinitionEntity;
 import org.camunda.bpm.engine.impl.pvm.process.ActivityImpl;
 import org.camunda.bpm.engine.impl.pvm.process.ScopeImpl;
 import org.camunda.bpm.engine.impl.util.xml.Element;
 import org.camunda.bpm.engine.management.Metrics;
+import org.camunda.bpm.engine.runtime.Execution;
 
 /**
  * @author Daniel Meyer
  *
  */
 public class MetricsBpmnParseListener extends AbstractBpmnParseListener {
 
-  public static MetricsExecutionListener ACTIVITY_INSTANCE_START_COUNTER = new MetricsExecutionListener(Metrics.ACTIVTY_INSTANCE_START);
-  public static MetricsExecutionListener ACTIVITY_INSTANCE_END_COUNTER = new MetricsExecutionListener(Metrics.ACTIVTY_INSTANCE_END);
+  public static MetricsExecutionListener ROOT_PROCESS_INSTANCE_START_COUNTER =
+      new MetricsExecutionListener(Metrics.ROOT_PROCESS_INSTANCE_START,
+                                   delegateExecution -> ((ExecutionEntity) delegateExecution)
+                                       .getRootProcessInstanceId()
+                                       .equals(delegateExecution.getId()));
```

Comment:
It's more robust to flip the operands of the comparison. An execution always has an ID, but an execution may not have a root process instance id (consider a migration case where a process instance was started in an old Camuna version without a root process instance id, and it executes a call activity in 7.13).